src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,175 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int INITIAL_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] nodes;
+    private int size;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
+        Node<K,V>[] currentNodes = nodes;
+        int hash = hash(key);
+        int length;
+        if (currentNodes == null || (length = currentNodes.length) == 0) {
+            currentNodes = resize();
+            length = currentNodes.length;
+        }
+        int idx = (length - 1) & hash;
+        Node<K,V> node1 = currentNodes[idx];
+        if (node1 == null) {
+            currentNodes[idx] = newNode(hash, key, value, null);
+        } else {
+            Node<K,V> node2;
+            K currentKey;
+            for (node2 = node1; node2 != null; node2 = node2.next) {
+                currentKey = node2.key;
+                if (node2.hash == hash && (Objects.equals(key, currentKey))) {
+                    node2.value = value;
+                    return;
+                }
+            }
+            node1.next = newNode(hash, key, value, node1.next);
+        }
 
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> node = getNode(key);
+        return node == null ? null : node.value;
     }
```

Comment:
Here is a hint how should your getValue method look like:

![image](https://github.com/mate-academy/jv-hash-map/assets/140905715/a2659f00-8343-46c9-80dc-b6b3146ad879)